lib/shopping_cart.rb

```diff
@@ -0,0 +1,58 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity.to_i
+    @products = []
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {}
+    details[:name] = @name
+    details[:capacity] = @capacity
+    details
+  end
+
+  def total_number_of_products
+    @products.sum do |product|
+      product.quantity
+    end
+  end
+
+  def is_full?
+    total_number_of_products >= @capacity
```

Comment:
I really like seeing that you're using a previous method to help with this functionality. 👍

Reply:
I think you just need this to be a > instead of a >= , according to the interaction pattern.